packages/stack/deployment/test/deployment.spec.js

```diff
@@ -0,0 +1,186 @@
+import sinon from 'sinon';
+import assert from 'assert';
+import Deployment from '../src/';
+
+const events = {
+  listeners: {},
+  setCommandHandler: sinon.spy((cmd, fn) => {
+    events.listeners[cmd] = fn;
+  }),
+  request: sinon.spy((cmd, ...args) => {
+    assert(events.listeners[cmd]);
+    events.listeners[cmd](...args);
+  }),
+  emit: (name) => {}
+};
+
+const plugins = {
+  listeners: {},
+  _plugin: {
+    listeners: {},
+    registerActionForEvent: (name, cb) => {
+      if (!plugins._plugin.listeners[name]) {
+        plugins._plugin.listeners[name] = [];
+      }
+      plugins._plugin.listeners[name].push(cb);
+    }
+  },
+  createPlugin: (name, opts) => {
+    return _plugin;
+  },
+  emitAndRunActionsForEvent: sinon.spy((name, options, cb) => {
+    if (plugins._plugin.listeners[name]) {
+      plugins._plugin.listeners[name].forEach(fn => {
+        fn(options, cb);
+      });
+    }
+  })
+};
+
+const fakeEmbark = {
+  events,
+  logger: {
+    info: () => {}
+  },
+  config: {
+    blockchainConfig: {}
+  },
+  assert: {
+    hasCommandHandler: (cmd) => {
+      sinon.assert.calledWith(fakeEmbark.events.setCommandHandler, cmd);
+    },
+    hasRequested: (cmd) => {
+      sinon.assert.calledWith(fakeEmbark.events.request, cmd);
+    },
+    hasRequestedWith: (cmd, ...args) => {
+      sinon.assert.calledWith(fakeEmbark.events.request, cmd, ...args);
+    },
+  },
+  teardown: () => {
+    fakeEmbark.events.listeners = {};
+    plugins._plugin.listeners = {};
+  }
+};
```

Comment:
The entire `fakeEmbark` construct including `plugins` and `events` will go away once #1894 lands.